experimental/rocm/tracing.c

```diff
@@ -128,6 +157,23 @@ iree_status_t iree_hal_rocm_tracing_context_allocate(
         timestamp_period);
   }
 
+  context->smutrace_enabled = false;
+  void (*dl) = dlopen(SMUTRACE_DLL, RTLD_LAZY);
+  if (dl) {
+    context->f_smuDumpInit = (SmuDumpInitFunc) dlsym(dl, "smuDumpInit");
+    context->f_smuDumpEnd = (SmuDumpEndFunc) dlsym(dl, "smuDumpEnd");
+    context->f_smuDumpOnce = (SmuDumpOnceFunc) dlsym(dl, "smuDumpOnce");
+    context->f_regDumpOnce = (RegDumpOnceFunc) dlsym(dl, "regDumpOnce");
+    context->f_sviDumpOnce = (SviDumpOnceFunc) dlsym(dl, "sviDumpOnce");
+    context->f_smuGetTraceRate = (SmuGetTraceRate) dlsym(dl, "getSmuVariablesCaptureRate");
+    context->f_regGetTraceRate = (RegGetTraceRate) dlsym(dl, "getRegisterExpressionCaptureRate");
+    context->smutrace_enabled = (context->f_smuDumpInit && context->f_smuDumpEnd && context->f_smuDumpOnce &&
+                        context->f_smuGetTraceRate && context->f_regGetTraceRate && context->f_regDumpOnce  && context->f_sviDumpOnce  &&
+                        context->f_smuDumpInit(iree_smutrace_callback));
+  } else {
+    fprintf(stderr, "Warning: " SMUTRACE_DLL " could not be loaded!\n");
```

Comment:
we don't use stderr (we're a library, not an application) - you can remove this or use a tracy logging macro like `IREE_TRACE_MESSAGE(WARNING, "whatever");`